src/java.base/share/classes/java/math/BigInteger.java

```diff
@@ -1581,7 +1584,30 @@ public BigInteger subtract(BigInteger val) {
      * @return {@code this * val}
      */
     public BigInteger multiply(BigInteger val) {
-        return multiply(val, false);
+        return multiply(val, false, false, 0);
+    }
+
+    /**
+     * Returns a BigInteger whose value is {@code (this * val)}.
+     * When both {@code this} and {@code val} are large, typically
+     * in the thousands of bits, parallel multiply might be used.
+     * This method returns the exact same mathematical result as
+     * {@link #multiply}.
+     *
+     * @implNote This implementation may offer better algorithmic
+     * performance when {@code val == this}.
+     *
+     * @implNote Compared to {@link #multiply}, this implementation's
+     * parallel multiplication algorithm will use more CPU resources
+     * to compute the result faster, with no increase in memory
+     * consumption.
```

Comment:
The implNote should cover a space of possible parallel multiply implementations so it doesn't have to be updated as often as the implementation is tuned or adjusted. So I'd prefer to have a statement like "may use more memory" even if the current implementation doesn't actually use more memory. If there are any "contraindications" on when to use the method, they could be listed here too.

Reply:
@kabutz I approved, but can you address Joe's comment, then i will update the CSR.

Reply:
Done.